addOneToPositiveNum(x) {
    if(x < 0) {
        throw ArgumentError("Argument x cannot be negative. Value of x is: ", x);
    }
    if(typeof x != int) {
        throw TypeError("Argument x must be of type int. Value of x is: ", x);
    }
    return x + 1;
}

try {
    print(addOneToPositiveNum(-1));
} catch {
    print("Something went wrong...");
}

try {
    print(addOneToPositiveNum(-1));
} catch (Error e) {
    print("Something went wrong...");
    print("Error message: ", e?.msg);
    print("Error arguments: ", e?.arg);
}

try {
    print(addOneToPositiveNum(-1));
} 
catch (ArgumentError e) {
    if (e?.arg typeof int) {
        print(addOneToPositiveNum(e?.arg * -1));
    }
}

try {
    print(addOneToPositiveNum());
} 
catch (ArgumentError | TypeError e) {
    print("There was an error with arguments");
    print("Error message: ", e?.msg);
    print("Error arguments: ", e?.arg);
}